src/neo4j_genai/generation/rag.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from typing import Optional, Any
+
+from pydantic import validate_call, ConfigDict
+
+from neo4j_genai.exceptions import LLMGenerationError
+from neo4j_genai.observers import ObserverInterface
+from neo4j_genai.retrievers.base import Retriever
+from neo4j_genai.generation.types import RagResultModel
+from neo4j_genai.generation.llm import LLMInterface
+from neo4j_genai.generation.prompts import RagTemplate
+from neo4j_genai.types import RetrieverResult
+
+logger = logging.getLogger(__name__)
+
+
+class RAG:
```

Comment:
Haven't thought more about the name, though keeping it simple is a good option to me, but open to other proposal.